frmts/georaster/cpl_vsil_ocilob.cpp

```diff
@@ -93,62 +92,101 @@ class VSIOCILobHandle : public VSIVirtualHandle
 // -----------------------------------------------------------------------------
 
-WSIOCILobFSHandle::~WSIOCILobFSHandle()
+char** WSIOCILobFSHandle::ParseIdentificator( const char* pszFilename )
 {
-    if( phLocator )
+    if( strncmp(pszFilename, "/vsiocilob/", strlen("/vsiocilob/") ) != 0 )
     {
-        OWStatement::Free( &phLocator, 1 );
+        return nullptr;
     }
 
-    if( poStatement )
-    {
-        delete poStatement;
-    }
+    char** papszParam = CSLTokenizeString2(
+                            &pszFilename[strlen("/vsiocilob/")], ",",
+                            CSLT_HONOURSTRINGS | CSLT_ALLOWEMPTYTOKENS |
+                            CSLT_STRIPLEADSPACES | CSLT_STRIPENDSPACES );
 
-    if( poConnection )
+    if( CSLCount( papszParam ) < 6 )
     {
-        delete poConnection;
+        CSLDestroy( papszParam );
+        return nullptr;
     }
+
+    return papszParam;
 }
 
 // -----------------------------------------------------------------------------
-//                                                          ParseIdentificator()
+//                                                                        GetConnection()
 // -----------------------------------------------------------------------------
 
-char** WSIOCILobFSHandle::ParseIdentificator( const char* pszFilename )
+OWConnection* WSIOCILobFSHandle::GetConnection(char** papszParam)
 {
-    if( strncmp(pszFilename, "/vsiocilob/", strlen("/vsiocilob/") ) != 0 )
+    OWConnection *poConnection = nullptr;
+
+    if( ! papszParam )
     {
         return nullptr;
     }
 
-    char** papszParam = CSLTokenizeString2(
-                            &pszFilename[strlen("/vsiocilob/")], ",",
-                            CSLT_HONOURSTRINGS | CSLT_ALLOWEMPTYTOKENS |
-                            CSLT_STRIPLEADSPACES | CSLT_STRIPENDSPACES );
+    if( strlen( papszParam[0] ) == 0 &&
+        strlen( papszParam[1] ) == 0 &&
+        strlen( papszParam[2] ) == 0 )
+    {
+        /* In an external procedure environment, before opening any
+         * dataset, the caller must pass the with_context as an
+         * string metadata item OCI_CONTEXT_PTR to the driver. */
+
+        OCIExtProcContext* with_context = nullptr;
+
+        const char* pszContext = GDALGetMetadataItem(
+                                           GDALGetDriverByName("GEORASTER"),
+                                          "OCI_CONTEXT_PTR", nullptr );
+
+        if( pszContext )
+        {
+            sscanf( pszContext, "%p", &with_context );
+
+            poConnection = new OWConnection( with_context );
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+    else
+    {
+        poConnection = new OWConnection( papszParam[0],
+                                         papszParam[1],
+                                         papszParam[2] );
+    }
 
-    if( CSLCount( papszParam ) < 6 )
+    if( ! poConnection->Succeeded() )
```

Comment:
poConnection should be deleted before return nullptr

Reply:
will fix this.